server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/copybooks/analysis/AbstractCopybookAnalysis.java

```diff
@@ -0,0 +1,344 @@
+import static org.eclipse.lsp.cobol.core.preprocessor.ProcessingConstants.*;
+
+/**
+ * This class is a framework for the copybook analysis. The actual implementations may change the
+ * behavior overriding the methods.
+ */
+@Slf4j
+abstract class AbstractCopybookAnalysis implements CopybookAnalysis {
+  protected static final int MAX_COPYBOOK_NAME_LENGTH_DEFAULT = Integer.MAX_VALUE;
+  private static final String HYPHEN = "-";
+  private static final String UNDERSCORE = "_";
+  private static final String SYNTAX_ERROR_CHECK_COPYBOOK_NAME =
+      "Syntax error by checkCopybookName: {}";
+
+  private final TextPreprocessor preprocessor;
+  private final CopybookService copybookService;
+  private final MessageService messageService;
+  private final int maxCopybookNameLength;
+
+  AbstractCopybookAnalysis(
+      TextPreprocessor preprocessor,
+      CopybookService copybookService,
+      MessageService messageService,
+      int maxCopybookNameLength) {
+    this.preprocessor = preprocessor;
+    this.copybookService = copybookService;
+    this.messageService = messageService;
+    this.maxCopybookNameLength = maxCopybookNameLength;
+  }
+
+  @Override
+  public PreprocessorFunctor handleCopybook(
+      ParserRuleContext context,
+      ParserRuleContext copySource,
+      CopybookConfig config,
+      String documentUri) {
+    return hierarchy -> {
+      List<SyntaxError> errors = new ArrayList<>();
+      CopybookMetaData metaData =
+          validateMetaData(maxCopybookNameLength)
+              .apply(
+                  CopybookMetaData.builder()
+                      .name(retrieveCopybookName(copySource))
+                      .context(context)
+                      .documentUri(documentUri)
+                      .copybookId(randomUUID().toString())
+                      .config(config)
+                      .nameLocality(
+                          LocalityUtils.buildLocality(
+                              copySource, documentUri, hierarchy.getCurrentCopybook()))
+                      .contextLocality(
+                          LocalityUtils.buildLocality(
+                              context, documentUri, hierarchy.getCurrentCopybook()))
+                      .build())
+              .unwrap(errors::addAll);
+
+      ExtendedDocument copybookDocument =
+          buildExtendedDocumentForCopybook(metaData).apply(hierarchy).unwrap(errors::addAll);
+      return stack -> {
+        writeText(metaData, copybookDocument).accept(stack);
+        return subContext -> {
+          storeCopyStatementSemantics(metaData, copybookDocument).accept(subContext);
+          return nestedMappings -> {
+            collectNestedSemanticData(metaData, copybookDocument).accept(nestedMappings);
+            return allErrors -> allErrors.addAll(errors);
+          };
+        };
+      };
+    };
+  }
+
+  private Function<CopybookMetaData, ResultWithErrors<CopybookMetaData>> validateMetaData(
+      int maxCopybookNameLength) {
```

Comment:
No way, man. You got `maxCopybookNameLength` as a private final field. There is no need to construct a new function.

Reply:
Okay...